### PR TITLE
PCHR-2379: Remove default filters from documents tab on contact page

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
This PR removes the default document filters by their status in contact page. Also removes the panel headings called "Ongoing" and "Resolved". And finally display all the documents in a list view in contact page.

## Before
Default document filters were present:
before--

## After
- Default document filters were removed and all documents are visible on load
- Removed panel title called "Ongoing" and "Resolved" and merged the ongoing and resolved section to display all the document list in single list view.
after--

## Technical Details
1.  To display status in filters, `$scope.filterParams.documentStatus` is used, so setting it to empty array so that we have no default status to filter the documents.
```js
$scope.filterParams = {
    contactId: null,
    documentStatus: [],
    ... 
};
```
2.  Remove following code form which was responsible for adding default  document status to filters.
```js
$scope.filterParams.documentStatus = $scope.filterParamsHolder.documentStatus;
```
form the applySidebarFilters function
```js
$scope.applySidebarFilters = function () {
      ...
}
```
3. Add to default document statuses to filter the documents in T&A dashboard in documents section.
```html
<tr
     ng-repeat="document in
      listPaginated = (listFiltered = (listOngoing = (list)
      ...
      | filterByStatus:filterParamsHolder.documentStatus
      ..."
      ng-controller="DocumentCtrl"
      ng-include
      src="'document.html'">
</tr>
```
4. Removes the markup from contact documents for  resolved and ongoing section.
```html
<h4 class="header-bar header-bar-danger">Resolved:
...
</h4>
<div uib-collapse="isCollapsed.documentListResolved">
...
</div>
```
5. Amend document filters to  filter all the documents including resolved document.
```html
<tr ng-repeat="document in listOngoing = (listFiltered = (list
    | filterByStatus:filterParams.documentStatus
    | filterByAssignmentType:filterParams.assignmentType
    | orderBy: 'activity_date_time'))"
    ng-controller="DocumentCtrl"
    ng-include src="'document.html'">
</tr>
```
6. Add `no-dropdown` attribute to fix style issue with multiple selection dropdown field as foe status filter field.
```html
<ui-select prevent-animations multiple ng-model="filterParams.documentStatus" no-dropdown>
```
Before: The selections are overlapped
screen shot 2017-08-04 at 2 41 51 pm

After: The overlapping style is removed.
screen shot 2017-08-04 at 2 40 25 pm

## Comments
- Added test cases for additional changes in removing filters from document  section in contact page

- [x] Tests Pass

### CODE REVIEW

**Comment by igorpavlov** [comment]
In the Before and After sections there are too much differences in the screenshots and I am not sure where should I pay attention to. Could you please provide GIFs or screenshots showing what exactly has been improved/fixed?

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js [hunk 1]
```diff
@@ -19,37 +19,28 @@ define([
...
-        { badgeClass: 'primary', calendarView: 'month', value: 'dueInNextFortnight' },
-        { badgeClass: 'primary', calendarView: 'month', value: 'dueInNinetyDays' }
-      ];
-
       $scope.dueToday = 0;
       $scope.dueThisWeek = 0;
       $scope.overdue = 0;
       $scope.list = documentList;
+      $scope.listResolvedLoaded = false;
+
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: Why was this moved and a line break added?

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js [hunk 2]
```diff
@@ -19,37 +19,28 @@ define([
 
       var defaultDocumentStatus = ['1', '2']; // 1: 'awaiting upload' | 2: 'awaiting approval
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: Are you sure `defaultDocumentStatus` is a good name for this variable. Since it is not used for showing default statuses I think it should be renamed.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js (hunk 1 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: Simply just to group the inline assignments, no big deal.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js (hunk 2 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: Well actually, `['1', '2']` are the status ids. So i guess it is more preferable to rename it as `deafaultStatusIds`. What do you say?

**Comment by bmnepali** [comment]
@igorpavlov I have update the Screenshots, to focus the areas where the fix is done. The fix is simply  removing the default document status form filter in contact page.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js (hunk 1 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: As discussed, we group by single line / multiline, then sort by property name.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js [hunk 3]
```diff
@@ -149,9 +161,16 @@ define([
...
       };
 
+      /**
+       * Creates the date range label using from and until dates in
+       * format 'dd/MM/yyyy'
+       *
+       * @param {string} from
+       * @param {string} until
+       */
       $scope.labelDateRange = function (from, until) {
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: Please add test cases to this function since there was a bug here and tests never failed. Please also check if there are another places apart from the controller (https://github.com/compucorp/civihr-tasks-assignments/pull/235/commits/d13d34ce133d08a0bd8b17748da1bbeaaa17d335#diff-c382f8565f58303b3569addc13e28686R136) and if they use from/until parameters. If there are such calls, we need to understand why everything worked before at all.

We may want to test this function with/without params.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js [hunk 4]
```diff
@@ -237,6 +286,7 @@ define([
         }
       });
 
+      // Init call
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: This comment is not needed.

Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/document-list.js]: done!

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/key-dates.html [hunk 5]
```diff
@@ -47,7 +47,7 @@ <h3 class="panel-title">View Dates By:</h3>
                             </div>
                         </div>
                         <button class="btn btn-default btn-primary btn-block"
-                                ng-click="labelDateRange(filterParams.dateRange.from, filterParams.dateRange.until);
+                                ng-click="labelDatZXCeRange(filterParams.dateRange.from, filterParams.dateRange.until);
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/key-dates.html]: Was this made by accident? As well, this function does not expect parameters now.

Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/key-dates.html]: ops! thanks for catching! also the indentation of the file should be fixed. Will do.

File: uk.co.compucorp.civicrm.tasksassignments/js/test/globals.js [hunk 6]
```diff
@@ -1,21 +1,34 @@
-(function (CRM) {
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/globals.js]: This is not how it should be implemented, here is a file by @reneolivo, he worked on a task which requires Drupal var.

https://github.com/civicrm/civihr/blob/523af796ce1c3105268a9f9aa9c34b0030bbf998/org.civicrm.reqangular/src/common/services/session.js

I suggest to copy his version and then @reneolivo will resolve conflicts by overriding this file with his one.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js [hunk 7]
```diff
@@ -10,15 +10,15 @@ define([
   'tasks-assignments/services/file',
   'tasks-assignments/services/dialog',
   'tasks-assignments/services/document',
-  'common/services/notification'
+  'common/services/notification.service'
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js]: Line 13 goes to line 8 as per https://compucorp.atlassian.net/wiki/spaces/PCHR/pages/121371880/AngularJS+Style+Guide

File: uk.co.compucorp.civicrm.tasksassignments/js/test/globals.js (hunk 6 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/globals.js]: Done!

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js (hunk 7 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js]: Done!

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/app-documents.js [hunk 8]
```diff
@@ -1,3 +1,6 @@
+/* globals angular */
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/app-documents.js]: inject `angular` via requirejs, don't declare it as global

**Comment by igorpavlov** [comment]
After section shows 2 solutions and I am not sure which one is the final, or maybe both are still valid. Please amend the After section with a GIF showing how filters finally work.

**Comment by bmnepali** [comment]
Fixed merge conflicts by rebase.
